packages/app/src/systems/Chains/eth/containers/ManageEthAssetsDialog.tsx

```diff
@@ -0,0 +1,176 @@
+                        : ''
+                    }`}
+                  </Form.ErrorMessage>
+                </Form.Control>
+              );
+            }}
+          />
+        </Box.Stack>
+        <CardList>
+          {/* TODO test that ust token works */}
+          {showUseTokenButton && (
+            <EthAssetCard
+              icon={
+                <Avatar.Generated
+                  size="xsm"
+                  hash={(assetInfo?.address || '') + (assetInfo?.symbol || '')}
+                />
+              }
+              name={assetInfo?.symbol || ''}
+              onAdd={() => {
+                handlers.addAsset({
+                  asset: {
+                    address: assetInfo?.address,
+                    decimals: assetInfo?.decimals,
+                    symbol: assetInfo?.symbol,
+                  },
+                });
+                store.openManageAssetsDialog();
+              }}
+            />
+          )}
+          {showCustomTokenButton && (
+            <EthAssetCard
+              icon={<Icon icon="HelpOctagon" />}
+              name={shortAddress(form.getValues('address'))}
+              onAdd={form.handleSubmit(onSubmitCustomToken)}
+            />
+          )}
+          {!(showCustomTokenButton || showUseTokenButton) &&
+            assets.map((asset, i) => (
+              <EthAssetCard
+                key={(asset.address || '') + (asset.symbol || '') + String(i)}
+                icon={
+                  asset.image ? (
+                    <Image alt=" " src={asset.image} />
+                  ) : (
+                    <Avatar.Generated
+                      size="xsm"
+                      key={
+                        (asset.address || '') + (asset.symbol || '') + String(i)
+                      }
+                      hash={
+                        (asset.address || '') + (asset.symbol || '') + String(i)
+                      }
+                    />
+                  )
+                }
+                name={asset.symbol || ''}
+                removeIconButton={
+                  asset.address !== NativeAssetId ? (
+                    <IconButton
+                      aria-label={(asset.address || '') + (asset.symbol || '')}
+                      variant="ghost"
+                      icon="SquareRoundedX"
+                      color="scalesRed10"
+                      onPress={() =>
+                        handlers.removeAsset({ address: asset.address })
+                      }
+                    />
+                  ) : (
+                    <IconButton
+                      aria-label={(asset.address || '') + (asset.symbol || '')}
+                      isDisabled
+                      tooltip="ETH is the native asset. It can't be removed."
+                      variant="ghost"
+                      icon="SquareRoundedX"
+                      color="scalesRed10"
+                    />
+                  )
+                }
```

Comment:
now I understand why you needed to pass `removeIconButton` to the component.

in this case you can:
- pass only `onRemove`
- render icon inside the component
- pass a prop `isRemoveDisabled` and `removeTooltip`